### PR TITLE
Prediction Guard LLM component

### PR DESCRIPTION
**Pull request by sharanshirodkar7**
## Description

This PR adds the LLMs component for Prediction Guard (PG) to the repository. The new component integrates various LLM capabilities to enhance the functionality of the Prediction Guard platform, allowing for more robust text generation and natural language processing features. The motivation behind this addition is to expand the library’s support for advanced language models and provide users with additional tools for their AI and ML workflows.

## Issues

n/a

## Type of change

-  New feature (non-breaking change which adds new functionality)

## Dependencies

PREDICTIONGUARD_API_KEY needs to be added to CI/CD to test the microservice.

## Tests

The following tests were run to verify the changes:

	•	Component functionality tests to ensure the new LLMs component works as expected.
	•	Integration tests to validate that the new component integrates seamlessly with the existing Prediction Guard ecosystem.
	•	Unit tests for each new function and module added as part of the LLMs component.

### CODE REVIEW

**Comment by sharanshirodkar7** [comment]
@chensuyue As per your suggestion I have closed the earlier joint components PR. This is a new PR just for PG LLM component. Kindly review. I will be creating separate components for embedding,LVMs and guardrails too. Thanks for the help.

Also did you add the PREDICTIONGUARD_API_KEY I provided you in the CI/CD. I see the failed check here because of no api key. Let me know.

File: comps/llms/text-generation/predictionguard/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+[Prediction Guard](https://docs.predictionguard.com) allows you to utilize hosted open access LLMs, LVMs, and embedding functionality with seamlessly integrated safeguards. In addition to providing a scalable access to open models, Prediction Guard allows you to configure factual consistency checks, toxicity filters, PII filters, and prompt injection blocking. Join the [Prediction Guard Discord channel](https://discord.gg/TFHgnhAFKd) and request an API key to get started.
+
+# Get Started
+
+## Build Docker Image
+
+```bash
+cd ../../..
+docker build -t opea/llm-predictionguard:latest -f comps/llms/text-generation/predictionguard/docker/Dockerfile .
```
Comment (ftian1) [inline comps/llms/text-generation/predictionguard/README.md:11]: pls note the docker image naming convention doesn't comply with the one we defined here

https://github.com/opea-project/docs/pull/107/files#diff-882936939f1eccccbb328c96070b53ef6c52b92f8403acb7f17618c368f47f67R92 line 92.

File: comps/llms/text-generation/predictionguard/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2024 Prediction Guard, Inc.
```
Comment (ftian1) [inline comps/llms/text-generation/predictionguard/docker/Dockerfile:1]: The content of this PR looks good to me. just a comment on the folder structure, we have refined it a bit to remove the 'docker' folder and flat all the files at a same parent folder. could you pls refer to the latest master and make corresponding changes?

**Comment by chensuyue** [comment]
Why there are 4 `.DS_Store`?

**Comment by sharanshirodkar7** [comment]
@chensuyue removed the .DS store files. The token is working, all checks passed. Ready for review. Thanks

**Review by lvliang-intel — APPROVED** [review]

**Review by chensuyue — APPROVED** [review]

**Comment by sharanshirodkar7** [comment]
@chensuyue @lvliang-intel thanks for the approvals. ready to merge.

**Pull request merged**